handlers/handler.go

```diff
@@ -54,8 +66,129 @@ type H struct {
 	config.Service
 }
 
+// GetUser retrieves the user based on information in the gin context.
+func (h *H) GetUser(ctx *gin.Context) (*model.User, *errors.Error) {
+	client, err := h.GetClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var name string
+	sess := h.Session(ctx)
+
+	username := sess.Get(SessionUsername)
+
+	// FIXME clean up this spaghetti -erikh
```

Comment:
Explain what you intend to change (and perhaps leave a github ticket link in the comment with more details) so if we don't get back to this in the near future we know what you were thinking at the time.

Reply:
Yeah cristian complained about this the last time it cropped up. I'll make a ticket.